resources/integration.rb

```diff
@@ -0,0 +1,57 @@
+# Install/remove a Datadog integration
+# This resource basically wraps the datadog-agent integration command to be able
+# to install and remove integrations from a Chef recipe.
+# The datadog_resource must be used on a system where the datadog-agent has
+# already been setup.
+
+default_action :install
+
+property :name, String, name_attribute: true
+
+property :version, String, required: true
+
+action :install do
+  unless node['datadog']['agent6']
+    log "The datadog_integration resource is only available with Agent v6." do
+      level :error
+    end
+    return
+  end
+
+  log "Getting integration #{new_resource.name}" do
+    level :debug
+  end
+
+  execute 'integration install' do
+    command   "\"#{agent_exe_filepath}\" integration install #{new_resource.name}==#{new_resource.version}"
+    user      'dd-agent' unless node['platform_family'] == 'windows'
+  end
+end
+
+action :remove do
+  unless node['datadog']['agent6']
+    log "The datadog_integration resource is only available with Agent v6." do
+      level :error
+    end
+    return
+  end
+
+  log "Removing integration #{new_resource.name}" do
+    level :debug
+  end
+
+  execute 'integration remove' do
+    command   "\"#{agent_exe_filepath}\" integration remove #{new_resource.name}"
```

Comment:
Is this "idempotent", or will calling the `integration remove` with a bad/removed integration name return a non-zero exit code?

Reply:
Good one! The `integration remove` return a 0 exit code if the integration is not installed so it's fine :+1: